transport/peer/peerlist/roundrobin.go

```diff
@@ -27,7 +27,9 @@ func NewRoundRobin(peerIDs []transport.PeerIdentifier, agent transport.PeerAgent
 	return rr, nil
 }
 
-type roundRobin struct {
+// RoundRobin is a mutable peerlist which contains a circular linked list of peers
+// to rotate through
+type RoundRobin struct {
 	sync.Mutex
```

Comment:
don't embed sync.Mutex in a public type if it's not part of the public interface for that type